src/libcore/intrinsics.rs

```diff
@@ -1941,6 +1941,16 @@ extern "rust-intrinsic" {
     ///
     /// Perma-unstable: do not use.
     pub fn miri_start_panic(payload: *mut u8) -> !;
+
+    /// See documentation of `<*const T>::guaranteed_eq` for details.
+    #[rustc_const_stable(feature = "const_raw_ptr_comparison", since = "1.46")]
+    #[cfg(not(bootstrap))]
+    pub fn ptr_guaranteed_eq<T>(ptr: *const T, other: *const T) -> bool;
```

Comment:
The reason I chose `guaranteed` instead of `maybe` (and invert the logic) is that I explained the `maybe` variant to someone and got recommended that I explain it this way, because it requires less negative reasoning.

Reply:
Seems reasonable, this also means our only example usecase does not need a negation. :)